launch/launch/launcher.py

```diff
@@ -225,7 +254,18 @@ def _run(self):
                         except ProcessLookupError:
                             pass
 
-            yield from asyncio.wait(all_futures.keys())
+            kwargs = {}
+            if not isinstance(threading.current_thread(), threading._MainThread):
+                # wake up periodically if we are not in the main thread
```

Comment:
Couldn't it be that launch itself is being using from a non `_MainThread`? Then this check wouldn't work.